### PR TITLE
Move checkout from pg to modal

### PR DESCRIPTION
**Pull request by tsyaeger**
Adds checkout as a modal in the profile page. Created a reusable modal component.

Also, I'm still having some trouble with responsiveness. Thought we could make it a separate issue.

### CODE REVIEW

File: client/src/components/payments/CheckoutForm.js [hunk 1]
```diff
@@ -84,21 +96,21 @@ class _CheckoutForm extends Component {
         if (this.state.investmentAmount >= 5) {
             this.setState({ investmentSaved: true })
         } else {
-            this.props.activateToast('Sorry, $5 minimum', 'neutral')
+            // this.props.activateToast('Sorry, $5 minimum', 'neutral')
+            console.log("no go")
```
Comment (evankp) [inline client/src/components/payments/CheckoutForm.js]: Little odd commenting out

File: client/src/pages/Project.js [hunk 2]
```diff
@@ -121,19 +163,6 @@ class Project extends React.Component {
...
-          userId: userDetails.id,
-          projectId: match.params.id,
-          projectTitle: this.state.project.title
-        }
-      })
-    };
-
     const disableFunding = () => {
       // TODO: Logic to handle disabling funding of project, maybe if project fund period has ended
       // Disabled by default for now until funding logic is added
```
Comment (evankp) [inline client/src/pages/Project.js]: Can remove todo

Comment (tsyaeger) [reply client/src/pages/Project.js]: You don't think this should still be done?

Comment (tsyaeger) [reply client/src/pages/Project.js]: if not, maybe I remove the function itself?

**Pull request merged**